src/library_pthread.js

```diff
@@ -849,11 +856,6 @@ var LibraryPThread = {
   // that the entire process should exit.
   // This function is always called from a pthread, but is executed on the
   // main thread due the __proxy attribute.
-  $exitOnMainThread__deps: ['exit',
-#if !MINIMAL_RUNTIME
-    '$handleException',
-#endif
-  ],
```

Comment:
Is this unrealted?

Reply:
This NFC change (commit 0f1095feefa0a50ff1dfe4d2494e4c33e5171ae7) was found while working on this, but is indeed unrelated. Let me know if you want to revert that too (for now).

Reply:
But this code calls the `_exit` C function so I don't understand how removing the dependency can be correct?

In any case please split this out into a separate change.

Reply:
These dependencies are also defined in `$PThread__deps`, so it's just de-duplicating the deps here.

https://github.com/emscripten-core/emscripten/blob/0f1095feefa0a50ff1dfe4d2494e4c33e5171ae7/src/library_pthread.js#L9-L18

I'll revert this change for now.